### ISSUE TITLE
aws_autoscaling_group tags and name do not support variable interpolation

### ISSUE THREAD

**Issue by mbainter** [hashicorp/terraform#3032]
With a terraform configuration like this:

```
resource "aws_autoscaling_group" "nat" {
  name = "${concat(var.vpc_name, \"-site\")}"
  launch_configuration = "${aws_launch_configuration.site.name}"
...
  tag {
    key = "Name"
    value = "${concat(\"Site for \", var.vpc_name)}"
    propagate_at_launch = true
  } 
} 
```

When I try to apply it sets the values as-is, instead of interpolating:

```
  tag.~3681847407.key:                 "" => "Name"
  tag.~3681847407.propagate_at_launch: "" => "1"
  tag.~3681847407.value:               "" => "${concat(\"Site for \", var.vpc_name)}"
```

The name likewise shows up incorrectly in AWS:

```
$ aws autoscaling describe-auto-scaling-groups --output=json | jq '.AutoScalingGroups[].AutoScalingGroupName'
"${var.vpc_name}-site"
```

**Comment by mbainter** [comment]
This can be closed.  The actual problem here was more complicated.

I created a VPC with a module, passing it a name for the VPC, as well as what region it should be created in.  I then created an "output" for name and region, but used aws_vpc.name and aws_vpc.region fo rthe values. Unfortunately, neither of those are available as outputs - but this doesn't throw an error.

The output from that was then used to set the module being called where this code is above, using something like:

```
module { 
  ...
  vpc_name = "${module.vpc.name}"
  region = "${module.vpc.region}"
  ...
}
```

Since these variables don't actually exist, the interpolation failed, resulting in the text being inserted instead.  I actually discovered this because my template in that same module was giving me errors about the variables being passed in not existing, when they were clearly defined.
